spacy/gold.pyx

```diff
@@ -216,9 +236,11 @@ class GoldCorpus(object):
         RETURNS (GoldCorpus): The newly created object.
         """
         self.limit = limit
+        self._max_sent_length = {'train': 0, 'dev': 0}
+        self._avg_sent_length = {'train': [0, 0, 0], 'dev': [0, 0, 0]}
```

Comment:
This feels a little bug-prone to me. Perhaps we should have a small datastructure instead, which keeps track of `max`, `len_sum`, `count` and `len2_sum` for one dataset, and then have one such datastructure for train and dev each ?